Dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM nixos/nix
+ADD . .
+RUN nix-channel --update
+CMD nix-shell --command nix-build default.nix
```

Comment:
Same question as in #111 : why run it in nix-shell? Also: `default.nix` is not necessary afaik.

Reply:
We can run it without shell as well but for users it is prefered to run within a shell so there arent traces of builds in the system. 
Using `default.nix` here to let `nix` build stuff and not cargo.

Reply:
But the `nix-shell` will probably use the `shell.nix` file, right? That seems unwanted since the build should be self-contained and work without that. Is this documented somewhere as a best practice?

Reply:
The build is self-contained, it is just considered a [good practice](https://nixos.wiki/wiki/Development_environment_with_nix-shell), though we can also test builds without shell.